src/OpenTelemetry.Exporter.OneCollector/Logs/OneCollectorLogExporterBuilder.cs

```diff
@@ -0,0 +1,118 @@
+            InstrumentationKey = instrumentationKey,
+        };
+    }
+
+    internal OneCollectorLogExporterBuilder(IConfiguration configuration)
+        : this(instrumentationKey: null)
+    {
+        Debug.Assert(configuration != null, "configuration was null");
+
+        configuration.Bind(this.Options);
+    }
+
+    internal OneCollectorLogExporterOptions Options { get; }
+
+    /// <summary>
+    /// Register a callback action for configuring the batch options of <see
+    /// cref="OneCollectorLogExporterOptions"/>.
+    /// </summary>
+    /// <param name="configure">Callback action for configuring <see
+    /// cref="BatchExportProcessorOptions{T}"/>.</param>
+    /// <returns>The supplied <see cref="OneCollectorLogExporterBuilder"/> for
+    /// call chaining.</returns>
+    public OneCollectorLogExporterBuilder ConfigureBatchOptions(Action<BatchExportProcessorOptions<LogRecord>> configure)
+    {
+        Guard.ThrowIfNull(configure);
+
+        configure(this.Options.BatchOptions);
+
+        return this;
+    }
+
+    /// <summary>
+    /// Register a callback action for configuring the transport options of <see
+    /// cref="OneCollectorLogExporterOptions"/>.
+    /// </summary>
+    /// <param name="configure">Callback action for configuring <see
+    /// cref="OneCollectorExporterTransportOptions"/>.</param>
+    /// <returns>The supplied <see cref="OneCollectorLogExporterBuilder"/> for
+    /// call chaining.</returns>
+    public OneCollectorLogExporterBuilder ConfigureTransportOptions(Action<OneCollectorExporterTransportOptions> configure)
+    {
+        Guard.ThrowIfNull(configure);
+
+        configure(this.Options.TransportOptions);
+
+        return this;
+    }
+
+    /// <summary>
+    /// Sets the <see cref="OneCollectorLogExporterOptions.DefaultEventName"/>
+    /// property. Default value: <c>Log</c>.
+    /// </summary>
+    /// <remarks><inheritdoc
+    /// cref="OneCollectorLogExporterOptions.DefaultEventName"
+    /// path="/remarks"/></remarks>
+    /// <param name="defaultEventName">Default event name.</param>
+    /// <returns>The supplied <see cref="OneCollectorLogExporterBuilder"/> for
+    /// call chaining.</returns>
+    public OneCollectorLogExporterBuilder SetDefaultEventName(string defaultEventName)
+    {
+        Guard.ThrowIfNullOrWhitespace(defaultEventName);
+
+        this.Options.DefaultEventName = defaultEventName;
+
+        return this;
+    }
+
+    /// <summary>
+    /// Sets the <see cref="OneCollectorExporterOptions.InstrumentationKey"/>
+    /// property.
+    /// </summary>
+    /// <remarks><inheritdoc
+    /// cref="OneCollectorExporterOptions.InstrumentationKey"
+    /// path="/remarks"/></remarks>
+    /// <param name="instrumentationKey">Instrumentation key.</param>
+    /// <returns>The supplied <see cref="OneCollectorLogExporterBuilder"/> for
+    /// call chaining.</returns>
+    public OneCollectorLogExporterBuilder SetInstrumentationKey(string instrumentationKey)
```

Comment:
Not blocking this PR - maybe use "ConnectionString" and get rid of these domain specific concepts (as they always change).

Reply:
Opened #1037